db/memtable.cc

```diff
@@ -316,7 +318,8 @@ class MemTableIterator : public InternalIterator {
   virtual void Seek(const Slice& k) override {
     PERF_TIMER_GUARD(seek_on_memtable_time);
     PERF_COUNTER_ADD(seek_on_memtable_count, 1);
-    if (bloom_ != nullptr) {
+    if (bloom_ && prefix_extractor_) {
```

Comment:
Since `bloom_` is a member of `MemTableIterator`. There is no need to set bloom_ if prefix_extractor_ is empty. In that case, we can void the prefix_extractor_ check here. On the other word, moving it to when the iterator is created.

Reply:
the check is already there in the constructor(line 288) so I'm going to simply remove the check here